### PR TITLE
Fix and Refine Language Injection Queries for vim.cmd, exec_lua, and Tree-sitter Queries

### PR DESCRIPTION
**Pull request by Shrutie03**
This update refines the Tree-sitter injection queries in runtime/queries/lua/injections.scm to improve handling of mixed language code within vim.cmd, exec_lua, and Tree-sitter query functions. The following changes are included:

Added #set! injection.combined to prevent comment leakage and scope language injection properly for multi-line strings.
Improved precision in detecting language injections for vim.api, vim.rpcrequest, and vim.api.nvim_create_autocmd to ensure accurate syntax highlighting and commentstring behavior.
Addressed issues with vim.keymap.set, vim.api.nvim_buf_create_user_command, and similar function calls. This resolves problems with incorrect comment strings in multi-language code blocks, especially when using embedded Lua, Vimscript, and Tree-sitter queries.

### CODE REVIEW

**Comment by clason** [comment]
Thank you, but these queries are maintained in nvim-treesitter. (And the queries we ship are intentionally minimal, omitting all injections that don't work out of the box.)

File: runtime/queries/lua/injections.scm [hunk 1]
```diff
@@ -11,23 +12,29 @@
...
 
+; Injection for Tree-sitter query functions (e.g., `vim.treesitter.query.set`) 
+; Added combined to properly handle multi-line strings.
 ((function_call
   name: (_) @_vimcmd_identifier
   arguments: (arguments
     (string
       content: _ @injection.content) .))
   (#set! injection.language "query")
+  (#set! injection.combined) ; Ensures query injection is isolated
```
Comment (clason) [inline runtime/queries/lua/injections.scm:34]: That actually ensures the exact opposite...

**Comment by justinmk** [comment]
@Shrutie03 can you provide some sort of response that suggests you are not a bot?

**Comment by Shrutie03** [comment]
Hi Justin,

I hope this message finds you well. I wanted to reach out and apologize for
my previous response—I realize it may not have been the most clear or
timely.

As my current contribution is intended for a college submission, I was
hoping you could assist me with some small changes in the code to ensure
everything is in order before I present it. Your expertise would be
incredibly helpful in making these final adjustments, and I truly
appreciate any guidance you can provide.

Once again, I apologize for any inconvenience, and I look forward to your
support.

Thank you so much for your time!

  Best regards,
  Shrutie03.

**Comment by Shrutie03** [comment]
Hi Cristian,

You’re absolutely right—it is a mandatory OS contribution for my
curriculum. I’ve been trying my best to understand the requirements, but
unfortunately, I’m running out of time for the submission, and I still feel
a bit lost. I’ve also been utilizing AI tools to help me along the way.

If you have any chance to assist me with some small contributions or
guidance, I would greatly appreciate it. Your expertise would really help
me get back on track.

Thank you for your understanding!

Best,
Shrutie03

**Comment by Shrutie03** [comment]
> Looks more like a mandatory OS contribution in some curriculum to me...

Hi Cristian,

You’re absolutely right—it is a mandatory OS contribution for my
curriculum. I’ve been trying my best to understand the requirements, but
unfortunately, I’m running out of time for the submission, and I still feel
a bit lost. I’ve also been utilizing AI tools to help me along the way.

If you have any chance to assist me with some small contributions or
guidance, I would greatly appreciate it. Your expertise would really help
me get back on track.

Thank you for your understanding!

Best regards,
Shrutie03

**Comment by Shrutie03** [comment]
> @Shrutie03 can you provide some sort of response that suggests you are not a bot?

Hi Justin,

I hope this message finds you well. I wanted to reach out and apologize for
my previous response—I realize it may not have been the most clear or
timely.

As my current contribution is intended for a college submission, I was
hoping you could assist me with some small changes in the code to ensure
everything is in order before I present it. Your expertise would be
incredibly helpful in making these final adjustments, and I truly
appreciate any guidance you can provide.

Once again, I apologize for any inconvenience, and I look forward to your
support.

Thank you so much for your time!

  Best regards,
  Shrutie03.

**Pull request closed without merging**